### ISSUE TITLE
GKE support for node-pools with SSD boot disk

### ISSUE THREAD

**Issue by JeanMertz** [kubernetes/kubernetes#36499]
Currently, using GKE, you can only use SSD's as attached disks, which can then be used for storage when using `hostPath` volumes.

It would be nice if I could create a node-pool that uses SSDs as the boot disk. This way, all running pods that don't use any external storage (or hostPath SSDs) can benefit from improved disk IO.

_I couldn't find any issue related to this, but that might be because my search term was too generic and I had to weed through hundreds of other issues_

**Comment by msau42** [comment]
We're planning on adding support for PD-SSD in GKE soon.

We can look into local SSD afterwards if there is interest.

cc @mindprince

**Comment by maikelvl** [comment]
@msau42 Hi, we are very interested in local SSD's? We're experiencing slow servers with standaard HDD boot disks.

**Comment by msau42** [comment]
@maikelvl will PD-SSD boot disks work for you?

**Comment by maikelvl** [comment]
@msau42 Unfortunately not. Those are only to be mounted into the Pods. We need a faster filesystem in all of Docker.

**Comment by msau42** [comment]
I mean, that if PD-SSD is the boot disk, then the entire node will be using it, including Docker.

**Comment by maikelvl** [comment]
OK. Yes, that would be great!

**Comment by svengt** [comment]
@msau42 I was wondering what the status is on PD-SSD boot disks support. Do you have any updates for us?

**Comment by msau42** [comment]
Hi @svengt, could you send me an email?  I would like to discuss your specific use cases and requirements further.

**Comment by motymichaely** [comment]
@msau42 We'd like to have PD-SSD boot disks support as well. Our use case is having large ephemeral pod disks (we can't really use hostPath without manually delete the hostPath content when pod is released, and we'd like to avoid this).

**Comment by msau42** [comment]
Hi @motymichaely, and anyone else interested in this feature,

It is now available via whitelist, so please send me an email if you want to be added to it.

**Comment by paolomainardi** [comment]
@msau42 please can you add me in the whitelist ?

**Comment by msau42** [comment]
Hi @paolomainardi please send me an email.  There are some forms to fill out.

**Comment by yuvipanda** [comment]
I'd like this too :D Where can I send you an email, @msau42?

**Comment by msau42** [comment]
@yuvipanda check my github profile

**Comment by sbeckwithiii** [comment]
I am thrilled to learn that GKE will be supporting PD-SSD soon!

**Comment by juanferreras** [comment]
@msau42 Hi, we'd be interested in getting whitelisted as well. I've tried emailing you but it bounced with a NoSuchUser error. Would you mind emailing me at juan@flioh.com? Thank you!

**Comment by halbritt** [comment]
Is this still outstanding?

**Comment by msau42** [comment]
Yes unfortunately it's still whitelist-only for the time-being.  Please contact me if you want to be added to the whitelist.

**Comment by lawrencedudley** [comment]
How are we looking on this? HDD feels particularly unsuited to container workloads :(

**Comment by halbritt** [comment]
I pinged @msau42 who got me on the whitelist for SSD boot disks.  That seemed to address my issues nicely.  One assumes that any container workload with a great deal of disk IO also requires persistence in which case one would use a PVC for that purpose.  All of my stateful workloads are on PVCs backed by SSD persistent disks.

The issue that I had is related to the fact that each pod gets ephemeral storage, kubelet monitors usage of that storage with a process that executes df and find very frequently.  If one has 100 pods per node, that can end up being a lot of inodes to traverse in very little time.  The end result will cause the node to go into notready state.  Since I switched to SSD boot drives, this problem has gone away.

**Comment by irees** [comment]
I've come back to this issue a few time recently. My work would also benefit from better I/O without the complexity of using a local-ssd via hostPath or persistent volume claims to use a PD-SSD.

PD-SSD boot disk would be great; mounting the local-ssd as /var/lib/docker would be even better.

**Comment by msau42** [comment]
@irees please send me an email and I can send you some forms to get whitelisted for PD-SSD boot disk
